### ISSUE TITLE
Is it possible to link libpcre statically with libyang?

### ISSUE THREAD

**Issue by swathins079** [CESNET/libyang#373]
I want to make a single binary in which libyang is statically linked to it. Is it possible? How can it be done.

Regards,
Swathin

**Comment by michalvasko** [comment]
Hi Swathin,
this is not really a question regarding *libyang* but rather a general Linux one. All I can suggest is for you to look up all the information you need on the Internet. For starters, you will need a static *libpcre* library, which you will likely have to compile yourself. If you then correctly modify how *libyang* is built, it should work.

Regards,
Michal

**Comment by mislavn** [comment]
I had some experience with it, you'll need to manually download and install libpcre from this [link](https://github.com/vmg/libpcre) and compile it statically.

> To build PCRE using static libraries only you must use --disable-shared when
> configuring it. For example:
> 
> ./configure --prefix=/usr/gnu --disable-shared
> 
> Then run "make" in the usual way. Similarly, you can use --disable-static to
> build only shared libraries.

If you wish to make a static libyang library it's not possible because libyang is using plugins.

Regards,
Mislav

**Comment by swathins079** [comment]
Hi,

Thanks Mislav, that really helped.

I was able to successfully link the libpcre to libyang. I did the following few steps to make it work.
Create a static library of libpcre with utf support(I got some error because of this being turned off), disable shared (as Mislav said). Then in the cmakelists.txt, replace the place where it looks for libpcre to static linking. By default, cmake will look for dynamic libs.

 `target_link_libraries(yang "-Wl,-Bstatic -L<path> -lpcre -Wl,-Bdynamic")`

-Wl,-Bdynamic can be replaced with the flag. Check the CMake documentation for help.

Hope this will help someone.

Regards,
Swathin
